app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  after_save :update_user_posts_counter
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] Great job leveraging Rails Active Callbacks! However, please note that the `update_user_posts_counter` method will be invoked each time a new post is saved to the database due to its callback association. To enhance code organization and encapsulation, it would be advisable to make the method private. By designating it as a private method, you ensure that it can only be called from within the class, preventing external access and promoting encapsulation of the logic. This helps maintain a more robust and maintainable codebase.
- [ ] This comment also applies to the `update_post_likes_counter` and `update_post_comments_counter` methods as they exhibit a similar pattern, keep up the great work! 💪

Reply:
Hi @AbelG101 , Thank you for reviewing my project 
i can not understand this parts
you mean that i should remove the after_save :update_user_posts_counter. and 
put private for all methods.
Please can you help me

Reply:
Hello @fatima-najafi sorry for the late reply, NO you don't need to remove anything just add the keyword `private` before the methods that's all.
Here's how it should look
```rb
private
def recent_post
  posts.order('created_at Desc').limit(3)
end
```

Reply:
thank you